csrc/expr_simplifier.cpp

```diff
@@ -1919,6 +1958,21 @@ Val* eliminateTrivialComputation(Val* value, const Context& context) {
       if (rhs->isOneInt()) {
         return IrBuilder::create<Val>(0L, *value->getDataType());
       }
+      // a % b -> a  if -|b| < a < |b|
+      Val* absrhs = foldConstants(IrBuilder::absExpr(rhs));
+      Val* negabsrhs = foldConstants(IrBuilder::negExpr(absrhs));
+      if (prove::lessThan(lhs, absrhs, context) &&
+          prove::lessThan(negabsrhs, lhs, context)) {
+        return lhs;
+      }
+    } else if (bop->getBinaryOpType() == BinaryOpType::Div) {
+      // a / b -> 0  if -|b| < a < |b|
+      Val* absrhs = foldConstants(IrBuilder::absExpr(rhs));
+      Val* negabsrhs = foldConstants(IrBuilder::negExpr(absrhs));
+      if (prove::lessThan(lhs, absrhs, context) &&
+          prove::lessThan(negabsrhs, lhs, context)) {
+        return IrBuilder::create<Val>(0L, *value->getDataType());
+      }
```

Comment:
This change seems to have altered how we simplify `0 / i0` (with no assumptions). Without this change, this is simplified to zero, ignoring the 0/0 case. After this change, it is not simplified.